packages/sitecore-jss/src/native-fetcher.test.ts

```diff
@@ -65,14 +86,38 @@ describe('NativeDataFetcher', () => {
   });
 
   describe('fetch', () => {
+    it('should execute request with fetch method', async () => {
+      const fetcher = new NativeDataFetcher();
+
+      spy.on(global, 'fetch', mockFetch(200));
+
+      const response = await fetcher.fetch('http://test.com/api');
+      expect(response.status).to.equal(200);
+      expect(fetchInput).to.equal('http://test.com/api');
+      expect(fetchInit?.method).to.equal('GET');
+      expect(fetchInit?.body).to.be.undefined;
+    });
+
+    it('should execute request with text response type', async () => {
+      const fetcher = new NativeDataFetcher();
+
+      spy.on(global, 'fetch', mockFetch(200, {}, { responseType: 'text' }));
+
+      const response = await fetcher.fetch('http://test.com/api');
+      expect(response.status).to.equal(200);
+      expect(fetchInput).to.equal('http://test.com/api');
+      expect(fetchInit?.method).to.equal('GET');
+      expect(fetchInit?.body).to.be.undefined;
+    });
+
     it('should execute POST request with data', async () => {
       const fetcher = new NativeDataFetcher();
       const postData = { x: 'val1', y: 'val2' };
       const respData = { z: 'val3' };
 
       spy.on(global, 'fetch', mockFetch(200, respData));
 
-      const response = await fetcher.fetch('http://test.com/api', postData);
```

Comment:
I think the idea here was to test if `fetch` switches between GET and POST requests, based on input.
`post` method test should be separate

Reply:
I wonder how it was working before because the `postData` doesn't conform to the expected `RequestInit` type in `fetch`
We need to use the post method which accepts `postData` with body param.

Reply:
True, previous implementation did not pass data as RequestInit type, instead a separate method prepared the request init object for `fetch` :
https://github.com/Sitecore/jss/blob/8f30b87b7f4ee14b37224c5858f072f21a130549/packages/sitecore-jss/src/native-fetcher.ts#L97C13-L97C27

Is the data type change neccessary?